functions/New-DbaDatabase.ps1

```diff
@@ -0,0 +1,469 @@
+		    [DbaInstanceParameter[]]$SqlInstance
+          , [PSCredential]$SqlCredential  
+          , [parameter(Mandatory = $true)]
+            [string]$DatabaseName                      
+          , [double]$NumberOfFilesInUserFilegroup = 1
+
+    # optional db variables
+          , [boolean]$UseDefaultFileLocations = $true
+          , [string]$NonDefaultFileLocation
+          , [string]$NonDefaultLogLocation
+          , [string]$Collation
+          , [string]$RecoveryModel
+          , [string]$DatabaseOwner = "sa"
+
+    #set the user data size, maxsize and growth
+          , [double]$UserDataFileSize 
+          , [double]$USerDataFileMaxSize
+          , [double]$UserDataFileGrowth #use 0 for no growth
+
+    #set the log size and growth
+          , [double]$LogSize
+          , [double]$LogGrowth
+
+    #set the primary file size in MB (will be converted to kb later)
+          , [double]$PrimaryFileSize = 10
+          , [double]$PrimaryFileGrowth = 10
+          , [double]$PrimaryFileMaxSize = 100
+          		
+    #Switches
+          , [switch]$Force
+		  , [switch]$Silent
+    )
+
+    begin
+    {
+        #Check file directories passed in if not using defaults
+        if ($UseDefaultFileLocations -eq $false -and ($NonDefaultFileLocation -eq $Null -or $NonDefaultLogLocation -eq $Null))
+            {
+               Stop-Function -Message "Non Default file locations selected, but are not supplied" -Category InvalidData -ErrorRecord $_
+               return
+            }
+
+        #Check that the user data file max size is greater than the user data file size
+        if ($UserDataFileSize -gt $UserDataFileMaxSize)
+        {
+            Stop-Function -Message "UserDataFilesize of $UserDataFileSize is greater than the UserFileMaxSize setting of $UserDataFileMaxSize" -Category InvalidData `
+            -ErrorRecord $_
+        }
+
+        #Check that the user data file max size is greater than the user file growth
+        if ($UserDataFileGrowth -gt $UserDataFileGrowth)
+        {
+            Stop-Function -Message "UserDataFileGrowth of $UserDataFileGrowth is greater than the UserFileMaxSize setting of $UserDataFileMaxSize" -Category InvalidData `
+            -ErrorRecord $_
+        }
+
+        #Check that the primary data file max size is greater than the primary data file size
+        if ($PrimaryFileSize -gt $PrimaryFileMaxSize)
+        {
+            Stop-Function -Message "PrimaryFilesize of $PrimaryFileSize is greater than the PrimaryFileMaxSize setting of $PrimaryFileMaxSize" -Category InvalidData `
+            -ErrorRecord $_
+        }
+
+        #Check that the user data file max size is greater than the user file growth
+        if ($PrimaryFileGrowth -gt $PrimaryFileGrowth)
+        {
+            Stop-Function -Message "PrimaryFileGrowth of $PrimaryFileGrowth is greater than the PrimaryFileMaxSize setting of $PrimaryFileMaxSize" -Category InvalidData `
+            -ErrorRecord $_
```

Comment:
There won't be any error record to pass in these checks. So if you want the function to stop need to add a return for each block. Unless your test showed otherwise?